crates/sui-indexer/src/indexer_reader.rs

```diff
@@ -217,6 +221,17 @@ impl IndexerReader {
         Ok(stored_epoch)
     }
 
+    pub fn get_latest_epoch_info_from_db(&self) -> Result<StoredEpochInfo, IndexerError> {
+        let stored_epoch = self.run_query(|conn| {
+            epochs::dsl::epochs
+                .order_by(epochs::epoch.desc())
+                .limit(1)
```

Comment:
nit: `first` covers `limit(1)`: https://docs.rs/diesel/latest/src/diesel/query_dsl/mod.rs.html#1782

same for other occurrences in this PR